torch/nn/modules/transformer.py

```diff
@@ -80,8 +80,16 @@ def __init__(self, d_model: int = 512, nhead: int = 8, num_encoder_layers: int =
 
         self.d_model = d_model
         self.nhead = nhead
-
+        self.num_encoder_layers = num_encoder_layers
+        self.num_decoder_layers = num_decoder_layers
+        self.dim_feedforward = dim_feedforward
+        self.dropout = dropout
+        self.activation = activation
+        self.custom_encoder = custom_encoder
+        self.custom_decoder = custom_decoder
+        self.layer_norm_eps = layer_norm_eps
         self.batch_first = batch_first
+        self.norm_first = norm_first
```

Comment:
This generates actual copies of layers just for bookkeeping, that later use resources for parameters when the model is moved to a GPU.  I'd have strong concerns about keeping modules in the architecture just for "book keeping"

Reply:
Maybe we can just keep those non-module attributes to save GPU memory, I think those non-module attributes should be saved.